offset/offset_clock_121_test.go

```diff
@@ -0,0 +1,106 @@
+//go:build go1.21
+
+package offset
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/vimeo/go-clocks/fake"
+)
+
+func TestOffsetClockContext121(t *testing.T) {
+	t.Run("ContextWithDeadlineCause", func(t *testing.T) {
+		base := time.Now()
+		inner := fake.NewClock(base)
+		c := NewOffsetClock(inner, time.Hour)
+
+		ctx, cancel := c.ContextWithDeadlineCause(context.Background(), inner.Now().Add(time.Hour), errors.New("test"))
+		t.Cleanup(cancel)
+
+		inner.Advance(2 * time.Hour)
+
+		select {
+		case <-ctx.Done():
+			if ctx.Err() != context.DeadlineExceeded {
+				t.Errorf("unexpected error: %v; expected %v", ctx.Err(), context.DeadlineExceeded)
+			}
+			if context.Cause(ctx) == nil || context.Cause(ctx).Error() != "test" {
+				t.Errorf("unexpected cause: %v; expected %v", context.Cause(ctx), "test")
+			}
+		case <-time.After(time.Second):
+			t.Errorf("context not done after 1 second")
+		}
+	})
+
+	t.Run("ContextWithDeadlineCauseCanceled", func(t *testing.T) {
+		base := time.Now()
+		inner := fake.NewClock(base)
+		c := NewOffsetClock(inner, time.Hour)
+
+		ctx, cancel := c.ContextWithDeadlineCause(context.Background(), inner.Now().Add(time.Hour), errors.New("test"))
+		t.Cleanup(cancel)
```

Comment:
Since you're calling `cancel` on the next line, I think you can skip this `t.Cleanup()` call.